pkg/config/linters_settings.go

```diff
@@ -828,6 +828,9 @@ type CustomLinterSettings struct {
 	Path string
 	// Description describes the purpose of the private linter.
 	Description string
-	// The URL containing the source code for the private linter.
+	// OriginalURL The URL containing the source code for the private linter.
 	OriginalURL string `mapstructure:"original-url"`
+
+	// Settings plugin settings only work with linterdb.PluginConstructor symbol.
+	Settings any
```

Comment:
On the other hand, referencing some external config file has the problem that the content of that file influences the linting result, but golangci-lint is unaware of that and thus may end up using obsolete cached results.

So perhaps not supporting it at all is the better choice?

In Kubernetes, I'll probably generate all config variants from a single .yaml.in with text templating and an `include` function for the logcheck config, in which case I won't need relative path support in golangci-lint.

But perhaps someone else will?

Reply:
`map[string]string` cannot be an alternative because it's too restrictive: it can be `map[string]any` or a slice (of struct/boolean/string/...).

An additional parameter with the configuration path can be possible but as you say we will have a problem with the cache.